diff --git a/index.js b/index.js
index cbd60d4687cd253558d57a422973f0c80a319c09..463d3c402acc08dd2fbd28269a82c3d6f5fc0ee7 100644
--- a/index.js
+++ b/index.js
@@ -3065,7 +3065,7 @@
 
     // First, split on pseudo-classes like :is, :where, etc. as we
     // don't want to split those by comma
-    var splitters = [/(?=\s:)/, /(?<=\)\)?[^,]*,)/];
+    var splitters = [/(?=\s:)/, /,(?=[^)]*$)/];
     var _loop = function _loop() {
       var splitter = _splitters[_i];
       selectors = selectors.flatMap(function (selector) {
diff --git a/index.mjs b/index.mjs
index f05c383a3e0ba383b9403e148d1f266bab61b231..fcf118a595fda986b28ae6d02ae11fce05a18b33 100644
--- a/index.mjs
+++ b/index.mjs
@@ -3059,7 +3059,7 @@ var parseCSSRuleForPseudos = function parseCSSRuleForPseudos(selectorText) {
 
   // First, split on pseudo-classes like :is, :where, etc. as we
   // don't want to split those by comma
-  var splitters = [/(?=\s:)/, /(?<=\)\)?[^,]*,)/];
+  var splitters = [/(?=\s:)/, /,(?=[^)]*$)/];
   var _loop = function _loop() {
     var splitter = _splitters[_i];
     selectors = selectors.flatMap(function (selector) {
