diff --git a/dist/chunk-3ZDIPBHM.mjs b/dist/chunk-3ZDIPBHM.mjs
index efe5a8debd5d314148a06bce5fd781aca5160ddc..3773e89812a792793d37340558979de6c98a6e11 100644
--- a/dist/chunk-3ZDIPBHM.mjs
+++ b/dist/chunk-3ZDIPBHM.mjs
@@ -44,7 +44,6 @@ var input = tv({
       "outline-none",
       "select-none",
       "opacity-0",
-      "hover:!opacity-100",
       "cursor-pointer",
       "active:!opacity-70",
       "rounded-full",
@@ -125,13 +124,13 @@ var input = tv({
       md: {
         inputWrapper: "h-10 min-h-10 rounded-medium",
         input: "text-small",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       },
       lg: {
         label: "text-medium",
         inputWrapper: "h-12 min-h-12 rounded-large",
         input: "text-medium",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       }
     },
     radius: {
diff --git a/dist/chunk-EYPT3KBI.mjs b/dist/chunk-EYPT3KBI.mjs
index 15663cbfe414824a4847bdfbd086c8ae224f3365..8e4ded5898c47273bb92549f06d1a14ace848a97 100644
--- a/dist/chunk-EYPT3KBI.mjs
+++ b/dist/chunk-EYPT3KBI.mjs
@@ -80,7 +80,7 @@ var breadcrumbItem = tv({
         item: "cursor-default"
       },
       false: {
-        item: ["hover:opacity-80", "active:opacity-disabled"]
+        item: ["hover:opacity-hover", "active:opacity-disabled"]
       }
     },
     isDisabled: {
diff --git a/dist/chunk-UEWXQXTA.mjs b/dist/chunk-UEWXQXTA.mjs
index 40fc49cd9673177fcf33c5418002269309c4ec1f..a814e933b3133fbe5d3398002ff89eb0d24e6c5a 100644
--- a/dist/chunk-UEWXQXTA.mjs
+++ b/dist/chunk-UEWXQXTA.mjs
@@ -46,7 +46,7 @@ var link = tv({
         "after:transition-background",
         "after:absolute"
       ],
-      false: "hover:opacity-80 active:opacity-disabled transition-opacity"
+      false: "hover:opacity-hover active:opacity-disabled transition-opacity"
     },
     isDisabled: {
       true: "opacity-disabled cursor-default pointer-events-none"
diff --git a/dist/components/breadcrumbs.js b/dist/components/breadcrumbs.js
index 37a2f2239052c7d5643f1e35c0c4de8c3cfe7c80..56d93991948decb1a1016422d2a23063d5629052 100644
--- a/dist/components/breadcrumbs.js
+++ b/dist/components/breadcrumbs.js
@@ -153,7 +153,7 @@ var breadcrumbItem = tv({
         item: "cursor-default"
       },
       false: {
-        item: ["hover:opacity-80", "active:opacity-disabled"]
+        item: ["hover:opacity-hover", "active:opacity-disabled"]
       }
     },
     isDisabled: {
diff --git a/dist/components/index.js b/dist/components/index.js
index 5fa42877de7564b782b1718556496eacab49e05a..60a3ba8930bac1225dcd883d54ce56ddd6eeed40 100644
--- a/dist/components/index.js
+++ b/dist/components/index.js
@@ -626,7 +626,7 @@ var link = tv({
         "after:transition-background",
         "after:absolute"
       ],
-      false: "hover:opacity-80 active:opacity-disabled transition-opacity"
+      false: "hover:opacity-hover active:opacity-disabled transition-opacity"
     },
     isDisabled: {
       true: "opacity-disabled cursor-default pointer-events-none"
@@ -4273,7 +4273,6 @@ var input = tv({
       "outline-none",
       "select-none",
       "opacity-0",
-      "hover:!opacity-100",
       "cursor-pointer",
       "active:!opacity-70",
       "rounded-full",
@@ -4354,13 +4353,13 @@ var input = tv({
       md: {
         inputWrapper: "h-10 min-h-10 rounded-medium",
         input: "text-small",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       },
       lg: {
         label: "text-medium",
         inputWrapper: "h-12 min-h-12 rounded-large",
         input: "text-medium",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       }
     },
     radius: {
@@ -8336,7 +8335,7 @@ var breadcrumbItem = tv({
         item: "cursor-default"
       },
       false: {
-        item: ["hover:opacity-80", "active:opacity-disabled"]
+        item: ["hover:opacity-hover", "active:opacity-disabled"]
       }
     },
     isDisabled: {
diff --git a/dist/components/input.js b/dist/components/input.js
index bdaa5a49dc3fca3391701248b8cc7b15d06c144a..d9df55b1497015e64397a29d23621df96287bc6d 100644
--- a/dist/components/input.js
+++ b/dist/components/input.js
@@ -125,7 +125,6 @@ var input = tv({
       "outline-none",
       "select-none",
       "opacity-0",
-      "hover:!opacity-100",
       "cursor-pointer",
       "active:!opacity-70",
       "rounded-full",
@@ -206,13 +205,13 @@ var input = tv({
       md: {
         inputWrapper: "h-10 min-h-10 rounded-medium",
         input: "text-small",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       },
       lg: {
         label: "text-medium",
         inputWrapper: "h-12 min-h-12 rounded-large",
         input: "text-medium",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       }
     },
     radius: {
diff --git a/dist/components/link.d.ts b/dist/components/link.d.ts
index bc40393ca311a2cc8dbb4a758af02124603f42c3..116d14da36e1f60a9715769bdc8779e391f205cc 100644
--- a/dist/components/link.d.ts
+++ b/dist/components/link.d.ts
@@ -31,7 +31,7 @@ declare const link: tailwind_variants.TVReturnType<{
     };
     isBlock: {
         true: string[];
-        false: "hover:opacity-80 active:opacity-disabled transition-opacity";
+        false: "hover:opacity-hover active:opacity-disabled transition-opacity";
     };
     isDisabled: {
         true: "opacity-disabled cursor-default pointer-events-none";
diff --git a/dist/components/link.js b/dist/components/link.js
index 517baccd4d957f86ec391ad88526c52a85bfb918..0b6f662852b3b612eba8952ecc286590b4363564 100644
--- a/dist/components/link.js
+++ b/dist/components/link.js
@@ -121,7 +121,7 @@ var link = tv({
         "after:transition-background",
         "after:absolute"
       ],
-      false: "hover:opacity-80 active:opacity-disabled transition-opacity"
+      false: "hover:opacity-hover active:opacity-disabled transition-opacity"
     },
     isDisabled: {
       true: "opacity-disabled cursor-default pointer-events-none"
diff --git a/dist/index.js b/dist/index.js
index 3b031a7453c6ed04ffdf4904e6377960086a76e7..81b00250c23814c87f2ea7a96cc5b8473970a00e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -688,7 +688,7 @@ var link = tv({
         "after:transition-background",
         "after:absolute"
       ],
-      false: "hover:opacity-80 active:opacity-disabled transition-opacity"
+      false: "hover:opacity-hover active:opacity-disabled transition-opacity"
     },
     isDisabled: {
       true: "opacity-disabled cursor-default pointer-events-none"
@@ -4335,7 +4335,6 @@ var input = tv({
       "outline-none",
       "select-none",
       "opacity-0",
-      "hover:!opacity-100",
       "cursor-pointer",
       "active:!opacity-70",
       "rounded-full",
@@ -4416,13 +4415,13 @@ var input = tv({
       md: {
         inputWrapper: "h-10 min-h-10 rounded-medium",
         input: "text-small",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       },
       lg: {
         label: "text-medium",
         inputWrapper: "h-12 min-h-12 rounded-large",
         input: "text-medium",
-        clearButton: "text-large"
+        clearButton: "text-large hover:!opacity-100"
       }
     },
     radius: {
@@ -8398,7 +8397,7 @@ var breadcrumbItem = tv({
         item: "cursor-default"
       },
       false: {
-        item: ["hover:opacity-80", "active:opacity-disabled"]
+        item: ["hover:opacity-hover", "active:opacity-disabled"]
       }
     },
     isDisabled: {
