diff --git a/dist/chunk-TTZQXKQZ.mjs b/dist/chunk-TTZQXKQZ.mjs
index f7f5c72f32e94213f48dc1d80186be536665a394..ac53f49ac4f6a1200e07d0e56de1708c5a524d48 100644
--- a/dist/chunk-TTZQXKQZ.mjs
+++ b/dist/chunk-TTZQXKQZ.mjs
@@ -54,7 +54,6 @@ var Tab = forwardRef((props, ref) => {
     rerender: true
   });
   const handleClick = () => {
-    chain(onClick, tabProps.onClick);
     if (!(domRef == null ? void 0 : domRef.current) || !(listRef == null ? void 0 : listRef.current))
       return;
     scrollIntoView(domRef.current, {
@@ -87,7 +86,7 @@ var Tab = forwardRef((props, ref) => {
           enabled: shouldFilterDOMProps,
           omitPropNames: /* @__PURE__ */ new Set(["title"])
         }),
-        { onClick: handleClick }
+        { onClick: chain(handleClick, onClick, tabProps.onClick) }
       ),
       className: (_a = slots.tab) == null ? void 0 : _a.call(slots, { class: tabStyles }),
       title: otherProps == null ? void 0 : otherProps.titleValue,
diff --git a/dist/index.js b/dist/index.js
index 8ff210aa74c8e7fc8162e252a5f0b4270121e83d..bb801909c9e0ede48fd9a374b91bd89ae270683d 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -206,7 +206,6 @@ var Tab = (0, import_system2.forwardRef)((props, ref) => {
     rerender: true
   });
   const handleClick = () => {
-    (0, import_utils2.chain)(onClick, tabProps.onClick);
     if (!(domRef == null ? void 0 : domRef.current) || !(listRef == null ? void 0 : listRef.current))
       return;
     (0, import_scroll_into_view_if_needed.default)(domRef.current, {
@@ -239,7 +238,7 @@ var Tab = (0, import_system2.forwardRef)((props, ref) => {
           enabled: shouldFilterDOMProps,
           omitPropNames: /* @__PURE__ */ new Set(["title"])
         }),
-        { onClick: handleClick }
+        { onClick: (0, import_utils2.chain)(handleClick, onClick, tabProps.onClick) }
       ),
       className: (_a = slots.tab) == null ? void 0 : _a.call(slots, { class: tabStyles }),
       title: otherProps == null ? void 0 : otherProps.titleValue,
diff --git a/dist/tab.js b/dist/tab.js
index 09588a6f0244285bb3efe34d07be53176e624faa..310bf00e6314d98e0342bed1325683cde516d2d6 100644
--- a/dist/tab.js
+++ b/dist/tab.js
@@ -83,7 +83,6 @@ var Tab = (0, import_system.forwardRef)((props, ref) => {
     rerender: true
   });
   const handleClick = () => {
-    (0, import_utils.chain)(onClick, tabProps.onClick);
     if (!(domRef == null ? void 0 : domRef.current) || !(listRef == null ? void 0 : listRef.current))
       return;
     (0, import_scroll_into_view_if_needed.default)(domRef.current, {
@@ -116,7 +115,7 @@ var Tab = (0, import_system.forwardRef)((props, ref) => {
           enabled: shouldFilterDOMProps,
           omitPropNames: /* @__PURE__ */ new Set(["title"])
         }),
-        { onClick: handleClick }
+        { onClick: (0, import_utils.chain)(handleClick, onClick, tabProps.onClick) }
       ),
       className: (_a = slots.tab) == null ? void 0 : _a.call(slots, { class: tabStyles }),
       title: otherProps == null ? void 0 : otherProps.titleValue,
diff --git a/dist/tabs.js b/dist/tabs.js
index 61765e340621ba9cda81e19b6ce612fe160fa747..876a91ae6afb13678136e13656de4ffe90546f7f 100644
--- a/dist/tabs.js
+++ b/dist/tabs.js
@@ -202,7 +202,6 @@ var Tab = (0, import_system2.forwardRef)((props, ref) => {
     rerender: true
   });
   const handleClick = () => {
-    (0, import_utils2.chain)(onClick, tabProps.onClick);
     if (!(domRef == null ? void 0 : domRef.current) || !(listRef == null ? void 0 : listRef.current))
       return;
     (0, import_scroll_into_view_if_needed.default)(domRef.current, {
@@ -235,7 +234,7 @@ var Tab = (0, import_system2.forwardRef)((props, ref) => {
           enabled: shouldFilterDOMProps,
           omitPropNames: /* @__PURE__ */ new Set(["title"])
         }),
-        { onClick: handleClick }
+        { onClick: (0, import_utils2.chain)(handleClick, onClick, tabProps.onClick) }
       ),
       className: (_a = slots.tab) == null ? void 0 : _a.call(slots, { class: tabStyles }),
       title: otherProps == null ? void 0 : otherProps.titleValue,
