diff --git a/common/lang.d.ts b/common/lang.d.ts
index 21e7232a8838fcde165597a45822e74688644476..0bd83066436d19f080fdd28855ebd635e2e8504a 100644
--- a/common/lang.d.ts
+++ b/common/lang.d.ts
@@ -1097,7 +1097,7 @@ declare module "../index" {
          * _.isObjectLike(null);
          * // => false
          */
-        isObjectLike(value?: any): boolean;
+        isObjectLike(value?: any): value is object;
     }
     interface LoDashImplicitWrapper<TValue> {
         /**
