diff --git a/dist/export/index.js b/dist/export/index.js
index 7cb007c1738889133a9a1d46f40fc6eb7d6d86a3..0688656f18353e76d2fc98137ae6ec09f5043e58 100644
--- a/dist/export/index.js
+++ b/dist/export/index.js
@@ -287,11 +287,7 @@ async function exportAppImpl(dir, options, span) {
             if ((routesManifest == null ? void 0 : (_routesManifest_rewrites = routesManifest.rewrites) == null ? void 0 : (_routesManifest_rewrites_beforeFiles = _routesManifest_rewrites.beforeFiles) == null ? void 0 : _routesManifest_rewrites_beforeFiles.length) > 0) {
                 const hasInterceptionRouteRewrite = routesManifest.rewrites.beforeFiles.some(_generateinterceptionroutesrewrites.isInterceptionRouteRewrite);
                 if (hasInterceptionRouteRewrite) {
-                    throw Object.defineProperty(new ExportError(`Intercepting routes are not supported with static export.\nRead more: https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features`), "__NEXT_ERROR_CODE", {
-                        value: "E626",
-                        enumerable: false,
-                        configurable: true
-                    });
+                    _log.warn(`Intercepting routes are not supported with static export.\nRead more: https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features`);
                 }
             }
             const actionIds = [
diff --git a/dist/server/dev/next-dev-server.js b/dist/server/dev/next-dev-server.js
index 3544cd86e4353c5c7b256b3af2fba0127bffc4bd..e15e2b46a04edfe6a77429b6355174a96fa89c24 100644
--- a/dist/server/dev/next-dev-server.js
+++ b/dist/server/dev/next-dev-server.js
@@ -465,8 +465,7 @@ class DevServer extends _nextserver.default {
     getinterceptionRoutePatterns() {
         const rewrites = (0, _generateinterceptionroutesrewrites.generateInterceptionRoutesRewrites)(Object.keys(this.appPathRoutes ?? {}), this.nextConfig.basePath).map((route)=>new RegExp((0, _buildcustomroute.buildCustomRoute)('rewrite', route).regex));
         if (this.nextConfig.output === 'export' && rewrites.length > 0) {
-            _log.error('Intercepting routes are not supported with static export.\nRead more: https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features');
-            process.exit(1);
+            _log.warn('Intercepting routes are not supported with static export.\nRead more: https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features');
         }
         return rewrites ?? [];
     }
